Classes/Controller/NewController.php

```diff
@@ -56,6 +56,16 @@ public function createAction(User $user)
         $user = FrontendUtility::forceValues($user, $this->config['new.']['forceValues.']['beforeAnyConfirmation.']);
         $user = UserUtility::fallbackUsernameAndPassword($user);
         $user = UserUtility::takeEmailAsUsername($user, $this->settings);
+    
+        $usernameValidation = $this->settings['new']['validation']['username'];
+        if ($usernameValidation['uniqueInDb'] || $usernameValidation['uniqueInPage']) {
+            if (count($this->userRepository->findByUsername($user->getUsername())) > 0) {
```

Comment:
This will only work correct for uniqueInDb. A check for uniqueInPage would require a page id.